Source/Core/Core/HW/GCMemcard/GCMemcard.cpp

```diff
@@ -1433,10 +1426,10 @@ s32 GCMemcard::FZEROGX_MakeSaveGameValid(const Header& cardheader, const DEntry&
   const auto [serial1, serial2] = cardheader.CalculateSerial();
 
   // set new serial numbers
-  *(u16*)&FileBuffer[1].m_block[0x0066] = BE16(BE32(serial1) >> 16);
-  *(u16*)&FileBuffer[3].m_block[0x1580] = BE16(BE32(serial2) >> 16);
-  *(u16*)&FileBuffer[1].m_block[0x0060] = BE16(BE32(serial1) & 0xFFFF);
-  *(u16*)&FileBuffer[1].m_block[0x0200] = BE16(BE32(serial2) & 0xFFFF);
```

Comment:
Type punning on assignment like that is undefined behavior, isn't it? Even if it isn't this looks pretty sketchy, maybe memcpy those bytes into the m_blocks instead?

Reply:
I'll save resolving this for a follow-up PR, given there's several other locations within the source file where this sort of type-punning is performed, and I'd rather resolve them all at once.

Reply:
Mhm, fair enough. Looks fine to me then.